Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.Cursor.cs

```diff
@@ -0,0 +1,148 @@
+        // public IntPtr Handle {get; set; }
+
+        // <summary>
+        // Gets the cursor hot spot.
+        // </summary>
+        // 
+        // <returns>
+        // A <see cref="T:System.Drawing.Point"/> representing the cursor hot spot.
+        // </returns>
+        // public Point HotSpot {get;}
+
+        // <summary>
+        // Gets or sets the cursor's position.
+        // </summary>
+        // 
+        // <returns>
+        // A <see cref="T:System.Drawing.Point"/> that represents the cursor's position in screen coordinates.
+        // </returns>
+        // public static Point Position {get; set;}
+
+        // <summary>
+        // Gets the size of the cursor object.
+        // </summary>
+        // 
+        // <returns>
+        // A <see cref="T:System.Drawing.Size"/> that represents the width and height of the <see cref="T:System.Windows.Forms.Cursor"/>.
+        // </returns>
+        // <filterpriority>1</filterpriority>
+        // public Size Size {get;}
+
+        // <summary>
+        // Gets or sets the object that contains data about the <see cref="T:System.Windows.Forms.Cursor"/>.
+        // </summary>
+        // 
+        // <returns>
+        // An <see cref="T:System.Object"/> that contains data about the <see cref="T:System.Windows.Forms.Cursor"/>. The default is null.
+        // </returns>
+        // <filterpriority>1</filterpriority>
+        // public object Tag {get; set;}
+
+
+        // <summary>
+        // Initializes a new instance of the <see cref="T:System.Windows.Forms.Cursor"/> class from the specified Windows handle.
+        // </summary>
+        // <param name="handle">An <see cref="T:System.IntPtr"/> that represents the Windows handle of the cursor to create. </param><exception cref="T:System.ArgumentException"><paramref name="handle"/> is <see cref="F:System.IntPtr.Zero"/>. </exception>
+        //public Cursor(IntPtr handle)
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:System.Windows.Forms.Cursor"/> class from the specified file.
+        /// </summary>
+        /// <param name="fileName">The cursor file to load. </param>
+        public Cursor(string fileName)
+        {
+            // This opens a stream, which will thrown an ArgumentNullException if FileName is null or empty.
+            Contract.Requires(!String.IsNullOrEmpty(fileName));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:System.Windows.Forms.Cursor"/> class from the specified resource with the specified resource type.
+        /// </summary>
+        /// <param name="type">The resource <see cref="T:System.Type"/>. </param><param name="resource">The name of the resource. </param>
+        public Cursor(System.Type type, string resource)
+        {
+            Contract.Requires(type != null || (type == null && resource != null));
```

Comment:
True. The intent was to capture that if type == null, then the resource can not be null.